src/Microsoft.Health.Abstractions/Data/ISink.cs

```diff
@@ -0,0 +1,31 @@
+﻿// -------------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License (MIT). See LICENSE in the repo root for license information.
+// -------------------------------------------------------------------------------------------------
+
+using System.Collections.Generic;
+using System.Threading.Tasks;
+
+namespace Microsoft.Health.Abstractions.Data
+{
+    /// <summary>
+    /// Generic Sink for writing data to any destination.
+    /// </summary>
+    /// <typeparam name="T">Generic data type T to write</typeparam>
+    public interface ISink<in T>
```

Comment:
As being mentioned in the meeting, let's rename these? like IEventSink?

Reply:
My idea was this is a shared component at Microsoft.Health level and not related to any particular domain (FHIR or Dicom) and thus allows any one of the products to use them. But renaming them under .Data would be misleading.

Reply:
Fixed with rename to IChangeFeedSource<T> and IChangeFeedSink<T>